File: src/02-challenge/solution.js
/**
 * @param {Array<{ toy: string, quantity: number }>} giftsToProduce
 * @returns {string[]} Array of manufactured gifts
 */
export default function fn(giftsToProduce) {
	/** @type {string[]} */
	const giftsArr = []

	for (const gift of giftsToProduce) {
		if (typeof gift.quantity !== 'number' || gift.quantity < 1) continue
		giftsArr.push(...Array.from({ length: gift.quantity }, () => gift.toy))
	}

	return giftsArr
}

File: src/02-challenge/solution.py
from typing import TypedDict

GiftToProduce = TypedDict(
    "GiftToProduce",
    {
        "toy": str,
        "quantity": int,
    },
)


def fn(gifts_to_produce: list[GiftToProduce]) -> list[str]:
    gifts: list[str] = []

    for gift in gifts_to_produce:
        if gift["quantity"] > 0:
            gifts.append(*[gift["toy"]] * gift["quantity"])

    return gifts
